### PR TITLE
Update summit modules

### PR DESCRIPTION
**Pull request by nkoukpaizan**
This PR updates the Summit modules.

### CODE REVIEW

**Review by cameronrutherford — APPROVED** [review]

**Comment by cameronrutherford** [comment]
Even with changes removing the `tests` and `logging` variant, we are still blocked because the non-mpi build needs to import `os`.

@nkoukpaizan if you want this merged ASAP, we can disable the `~mpi` build for now, and re-add once Spack PR is merged.

File: .github/workflows/spack_cpu_build.yaml [hunk 1]
```diff
@@ -13,13 +13,13 @@ jobs:
       matrix:
         spack_spec:
           # Minimal Build
-          - exago@develop~mpi~ipopt~hiop~logging~python~raja~tests
-          - exago@develop+mpi~ipopt+hiop~logging~python~raja~tests ^openmpi
+          - exago@develop~mpi~ipopt~hiop~python~raja
```
Comment (cameronrutherford) [inline .github/workflows/spack_cpu_build.yaml:16]:
```suggestion
           # See #18 
        # - exago@develop~mpi~ipopt~hiop~python~raja
```

**Review by cameronrutherford — APPROVED** [review]
#18

**Comment by cameronrutherford** [comment]
@ryandanehy or @nkoukpaizan please give this a quick review as GitHub is requiring an additional reviewer (since I made the latest changes).

**Comment by cameronrutherford** [comment]
Need to re-visit but it seems like Python CMake configuration is erroring when Python is not found. Simple problem but need to make sure we make a robust solution.

**Comment by cameronrutherford** [comment]
Spack PR was merged, so I think if we update the submodule to include these changes the pipelines will pass

**Comment by nkoukpaizan** [comment]
Python build is failing. I think we need to update the pybind11 submodule. This closed issue in their repository seems relevant: https://github.com/pybind/pybind11/issues/4464

**Comment by cameronrutherford** [comment]
@nkoukpaizan I am forever thankful that you were able to reproduce #23 :)

cc @jaelynlitz as she was going down a rabbit hole of updating the compiler to catch this bug, but it seems like we don't quite need to do that. @jaelynlitz please only modify compilers on CI platforms if you need to do so in order to upgrade C++ standard.

@nkoukpaizan I agree with your assessment that we need to change our pybind submodule to fix this. We need to also move pybind dep into spack eventually, but this is an easy thing to fix for the moment.

**Comment by cameronrutherford** [comment]
> @nkoukpaizan I am forever thankful that you were able to reproduce #23 :)
> 
> cc @jaelynlitz as she was going down a rabbit hole of updating the compiler to catch this bug, but it seems like we don't quite need to do that. @jaelynlitz please only modify compilers on CI platforms if you need to do so in order to upgrade C++ standard.
> 
> @nkoukpaizan I agree with your assessment that we need to change our pybind submodule to fix this. We need to also move pybind dep into spack eventually, but this is an easy thing to fix for the moment.

Since this fixes CI, I think this also closes #23

**Comment by jaelynlitz** [comment]
> @nkoukpaizan I am forever thankful that you were able to reproduce #23 :)
> 
> cc @jaelynlitz as she was going down a rabbit hole of updating the compiler to catch this bug, but it seems like we don't quite need to do that. @jaelynlitz please only modify compilers on CI platforms if you need to do so in order to upgrade C++ standard.
> 
> @nkoukpaizan I agree with your assessment that we need to change our pybind submodule to fix this. We need to also move pybind dep into spack eventually, but this is an easy thing to fix for the moment.

What exactly was the fix? updating pybind? I also see a magma upgrade from 2.6.2 to 2.7.1?

**Comment by cameronrutherford** [comment]
> > @nkoukpaizan I am forever thankful that you were able to reproduce #23 :)
> 
> > 
> 
> > cc @jaelynlitz as she was going down a rabbit hole of updating the compiler to catch this bug, but it seems like we don't quite need to do that. @jaelynlitz please only modify compilers on CI platforms if you need to do so in order to upgrade C++ standard.
> 
> > 
> 
> > @nkoukpaizan I agree with your assessment that we need to change our pybind submodule to fix this. We need to also move pybind dep into spack eventually, but this is an easy thing to fix for the moment.
> 
> 
> 
> What exactly was the fix? updating pybind? I also see a magma upgrade from 2.6.2 to 2.7.1?

Fix to Python error was pybind. Magma upgrade might cause testing errors, but these are summit modules not run in CI, so it's harder to track.

**Review by cameronrutherford — APPROVED** [review]

**Pull request merged**